src/care/ArrayView.h

```diff
@@ -0,0 +1,68 @@
+#ifndef CARE_ARRAY_VIEW_H
+#define CARE_ARRAY_VIEW_H
+
+#include "care/host_device_ptr.h"
+#include "chai/ExecutionSpaces.hpp"
+#include "RAJA/util/View.hpp"
+
+namespace care {
+   ///
+   /// Creates a 1D view on top of a flat allocation (non-owning semantics)
+   ///
+   /// Moves the data to the default execution space (determined at
+   /// configuration time) and wraps it in a RAJA::View.
+   ///
+   /// @param[in] data Container for the allocation
+   /// @param[in] extent Length of the view
+   ///
+   /// @return A RAJA::View object
+   ///
+   template <class T>
+   auto makeArrayView1D(care::host_device_ptr<T> data,
+                        int extent);
+
+   ///
+   /// Creates a 2D view on top of a flat allocation (non-owning semantics)
+   ///
+   /// Moves the data to the default execution space (determined at
+   /// configuration time) and wraps it in a RAJA::View. When using with
+   /// a RAJA loop, extent2 should correspond to the length of the RAJA
+   /// loop for optimal performance.
+   ///
+   /// @param[in] data Container for the allocation
+   /// @param[in] extent1 Length of the 1st dimension
```

Comment:
does there need to some description of what 1st and 2nd dimension are?

Reply:
I'm not sure what that description should be. I added the comment starting at line 27 to try to describe which length should go last in the argument list. If you have suggestions, I'm certainly open to providing more clarification.

Reply:
Yeah, I don't know either.

Reply:
I'm going to go ahead with merging this, but I'm happy to revisit when we start using this in applications and find that more clarification is needed.